### PR TITLE
Add the local (dev) CA cert to CA cert store of php container

### LINKED ISSUE #1405: Standard Edition misconfiguration
**API Platform version(s) affected**: 2.5.4

**Description**  
The certificates and .env are misconfigured

**How to reproduce**  
Install the standard edition, set the `Greetings` Resource with `mercure=true`.

**Possible Solution**  
* Add a test to ensure a POST on Greetings with mercure at true returns a 200.
* Add the MkCert certificates to the bundle
* Add the bundle to dev-tls
* Mount dev-certs in volume into php container
* Use the .pem file in the api-platform.dev.ini file

**Additional Context**  

<details>
  <summary>Click to the requests and responses</summary>

The Resource
```php
/**
 * @ApiResource(mercure=true|false)
 * @ORM\Entity
 */
class Greeting
{
    /**
     * @var int The entity Id
     *
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     */
    private $id;

    /**
     * @var string A nice person
     *
     * @ORM\Column
     * @Assert\NotBlank
     */
    public $name = '';

    public function getId(): int
    {
        return $this->id;
    }
}
```

Request:
```shell
curl --location --request POST 'https://localhost:8443/greetings' \
--header 'Content-Type: application/json' \
--header 'Accept: application/ld+json' \
--data-raw '{
	"name": "Vidal Larson"
}'
```

Response when mercure is at `false`: 
```json
{
    "@context": "/contexts/Greeting",
    "@id": "/greetings/1",
    "@type": "Greeting",
    "name": "Vidal Larson",
    "id": 1
}
```

Response when mercure is at `true`:
```json
{
    "@context": "/contexts/Error",
    "@type": "hydra:Error",
    "hydra:title": "An error occurred",
    "hydra:description": "Couldn't connect to server for \"http://mercure/.well-known/mercure\".",
    "trace": [
        {
            "namespace": "",
            "short_class": "",
            "class": "",
            "type": "",
            "function": "",
            "file": "/srv/api/vendor/symfony/http-client/Chunk/ErrorChunk.php",
            "line": 64,
            "args": []
        },
        {
            "namespace": "Symfony\\Component\\HttpClient\\Chunk",
            "short_class": "ErrorChunk",
            "class": "Symfony\\Component\\HttpClient\\Chunk\\ErrorChunk",
            "type": "->",
            "function": "isFirst",
            "file": "/srv/api/vendor/symfony/http-client/Response/CurlResponse.php",
            "line": 147,
            "args": []
        },
        {
            "namespace": "Symfony\\Component\\HttpClient\\Response",
            "short_class": "CurlResponse",
            "class": "Symfony\\Component\\HttpClient\\Response\\CurlResponse",
            "type": "::",
            "function": "Symfony\\Component\\HttpClient\\Response\\{closure}",
            "file": "/srv/api/vendor/symfony/http-client/Response/ResponseTrait.php",
            "line": 102,
            "args": [
                [
                    "object",
                    "Symfony\\Component\\HttpClient\\Response\\CurlResponse"
                ]
            ]
        },
        {
            "namespace": "Symfony\\Component\\HttpClient\\Response",
            "short_class": "CurlResponse",
            "class": "Symfony\\Component\\HttpClient\\Response\\CurlResponse",
            "type": "->",
            "function": "doGetContent",
            "file": "/srv/api/vendor/symfony/http-client/Response/CurlResponse.php",
            "line": 205,
            "args": [
                [
                    "boolean",
                    true
                ]
            ]
        },
        {
            "namespace": "Symfony\\Component\\HttpClient\\Response",
            "short_class": "CurlResponse",
            "class": "Symfony\\Component\\HttpClient\\Response\\CurlResponse",
            "type": "->",
            "function": "getContent",
            "file": "/srv/api/vendor/symfony/mercure/src/Publisher.php",
            "line": 61,
            "args": []
        },
        {
            "namespace": "Symfony\\Component\\Mercure",
            "short_class": "Publisher",
            "class": "Symfony\\Component\\Mercure\\Publisher",
            "type": "->",
            "function": "__invoke",
            "file": "/srv/api/vendor/symfony/mercure/src/Debug/TraceablePublisher.php",
            "line": 44,
            "args": [
                [
                    "object",
                    "Symfony\\Component\\Mercure\\Update"
                ]
            ]
        },
        {
            "namespace": "Symfony\\Component\\Mercure\\Debug",
            "short_class": "TraceablePublisher",
            "class": "Symfony\\Component\\Mercure\\Debug\\TraceablePublisher",
            "type": "->",
            "function": "__invoke",
            "file": "/srv/api/vendor/api-platform/core/src/Bridge/Doctrine/EventListener/PublishMercureUpdatesListener.php",
            "line": 184,
            "args": [
                [
                    "object",
                    "Symfony\\Component\\Mercure\\Update"
                ]
            ]
        },
        {
            "namespace": "ApiPlatform\\Core\\Bridge\\Doctrine\\EventListener",
            "short_class": "PublishMercureUpdatesListener",
            "class": "ApiPlatform\\Core\\Bridge\\Doctrine\\EventListener\\PublishMercureUpdatesListener",
            "type": "->",
            "function": "publishUpdate",
            "file": "/srv/api/vendor/api-platform/core/src/Bridge/Doctrine/EventListener/PublishMercureUpdatesListener.php",
            "line": 99,
            "args": [
                [
                    "object",
                    "App\\Entity\\Greeting"
                ],
                [
                    "array",
                    []
                ]
            ]
        },
        {
            "namespace": "ApiPlatform\\Core\\Bridge\\Doctrine\\EventListener",
            "short_class": "PublishMercureUpdatesListener",
            "class": "ApiPlatform\\Core\\Bridge\\Doctrine\\EventListener\\PublishMercureUpdatesListener",
            "type": "->",
            "function": "postFlush",
            "file": "/srv/api/vendor/symfony/doctrine-bridge/ContainerAwareEventManager.php",
            "line": 58,
            "args": [
                [
                    "object",
                    "Doctrine\\ORM\\Event\\PostFlushEventArgs"
                ]
            ]
        },
        {
            "namespace": "Symfony\\Bridge\\Doctrine",
            "short_class": "ContainerAwareEventManager",
            "class": "Symfony\\Bridge\\Doctrine\\ContainerAwareEventManager",
            "type": "->",
            "function": "dispatchEvent",
            "file": "/srv/api/vendor/doctrine/orm/lib/Doctrine/ORM/UnitOfWork.php",
            "line": 3412,
            "args": [
                [
                    "string",
                    "postFlush"
                ],
                [
                    "object",
                    "Doctrine\\ORM\\Event\\PostFlushEventArgs"
                ]
            ]
        },
        {
            "namespace": "Doctrine\\ORM",
            "short_class": "UnitOfWork",
            "class": "Doctrine\\ORM\\UnitOfWork",
            "type": "->",
            "function": "dispatchPostFlushEvent",
            "file": "/srv/api/vendor/doctrine/orm/lib/Doctrine/ORM/UnitOfWork.php",
            "line": 444,
            "args": []
        },
        {
            "namespace": "Doctrine\\ORM",
            "short_class": "UnitOfWork",
            "class": "Doctrine\\ORM\\UnitOfWork",
            "type": "->",
            "function": "commit",
            "file": "/srv/api/vendor/doctrine/orm/lib/Doctrine/ORM/EntityManager.php",
            "line": 368,
            "args": [
                [
                    "null",
                    null
                ]
            ]
        },
        {
            "namespace": "Doctrine\\ORM",
            "short_class": "EntityManager",
            "class": "Doctrine\\ORM\\EntityManager",
            "type": "->",
            "function": "flush",
            "file": "/srv/api/vendor/api-platform/core/src/Bridge/Doctrine/Common/DataPersister.php",
            "line": 59,
            "args": []
        },
        {
            "namespace": "ApiPlatform\\Core\\Bridge\\Doctrine\\Common",
            "short_class": "DataPersister",
            "class": "ApiPlatform\\Core\\Bridge\\Doctrine\\Common\\DataPersister",
            "type": "->",
            "function": "persist",
            "file": "/srv/api/vendor/api-platform/core/src/Bridge/Symfony/Bundle/DataPersister/TraceableChainDataPersister.php",
            "line": 56,
            "args": [
                [
                    "object",
                    "App\\Entity\\Greeting"
                ],
                [
                    "array",
                    {
                        "resource_class": [
                            "string",
                            "App\\Entity\\Greeting"
                        ],
                        "collection_operation_name": [
                            "string",
                            "post"
                        ],
                        "receive": [
                            "boolean",
                            true
                        ],
                        "respond": [
                            "boolean",
                            true
                        ],
                        "persist": [
                            "boolean",
                            true
                        ]
                    }
                ]
            ]
        },
        {
            "namespace": "ApiPlatform\\Core\\Bridge\\Symfony\\Bundle\\DataPersister",
            "short_class": "TraceableChainDataPersister",
            "class": "ApiPlatform\\Core\\Bridge\\Symfony\\Bundle\\DataPersister\\TraceableChainDataPersister",
            "type": "->",
            "function": "persist",
            "file": "/srv/api/vendor/api-platform/core/src/EventListener/WriteListener.php",
            "line": 76,
            "args": [
                [
                    "object",
                    "App\\Entity\\Greeting"
                ],
                [
                    "array",
                    {
                        "resource_class": [
                            "string",
                            "App\\Entity\\Greeting"
                        ],
                        "collection_operation_name": [
                            "string",
                            "post"
                        ],
                        "receive": [
                            "boolean",
                            true
                        ],
                        "respond": [
                            "boolean",
                            true
                        ],
                        "persist": [
                            "boolean",
                            true
                        ]
                    }
                ]
            ]
        },
        {
            "namespace": "ApiPlatform\\Core\\EventListener",
            "short_class": "WriteListener",
            "class": "ApiPlatform\\Core\\EventListener\\WriteListener",
            "type": "->",
            "function": "onKernelView",
            "file": "/srv/api/vendor/symfony/event-dispatcher/Debug/WrappedListener.php",
            "line": 126,
            "args": [
                [
                    "object",
                    "Symfony\\Component\\HttpKernel\\Event\\ViewEvent"
                ],
                [
                    "string",
                    "kernel.view"
                ],
                [
                    "object",
                    "Symfony\\Component\\HttpKernel\\Debug\\TraceableEventDispatcher"
                ]
            ]
        },
        {
            "namespace": "Symfony\\Component\\EventDispatcher\\Debug",
            "short_class": "WrappedListener",
            "class": "Symfony\\Component\\EventDispatcher\\Debug\\WrappedListener",
            "type": "->",
            "function": "__invoke",
            "file": "/srv/api/vendor/symfony/event-dispatcher/EventDispatcher.php",
            "line": 264,
            "args": [
                [
                    "object",
                    "Symfony\\Component\\HttpKernel\\Event\\ViewEvent"
                ],
                [
                    "string",
                    "kernel.view"
                ],
                [
                    "object",
                    "Symfony\\Component\\HttpKernel\\Debug\\TraceableEventDispatcher"
                ]
            ]
        },
        {
            "namespace": "Symfony\\Component\\EventDispatcher",
            "short_class": "EventDispatcher",
            "class": "Symfony\\Component\\EventDispatcher\\EventDispatcher",
            "type": "->",
            "function": "doDispatch",
            "file": "/srv/api/vendor/symfony/event-dispatcher/EventDispatcher.php",
            "line": 239,
            "args": [
                [
                    "array",
                    [
                        [
                            "object",
                            "Symfony\\Component\\EventDispatcher\\Debug\\WrappedListener"
                        ],
                        [
                            "object",
                            "Symfony\\Component\\EventDispatcher\\Debug\\WrappedListener"
                        ],
                        [
                            "object",
                            "Symfony\\Component\\EventDispatcher\\Debug\\WrappedListener"
                        ],
                        [
                            "object",
                            "Symfony\\Component\\EventDispatcher\\Debug\\WrappedListener"
                        ]
                    ]
                ],
                [
                    "string",
                    "kernel.view"
                ],
                [
                    "object",
                    "Symfony\\Component\\HttpKernel\\Event\\ViewEvent"
                ]
            ]
        },
        {
            "namespace": "Symfony\\Component\\EventDispatcher",
            "short_class": "EventDispatcher",
            "class": "Symfony\\Component\\EventDispatcher\\EventDispatcher",
            "type": "->",
            "function": "callListeners",
            "file": "/srv/api/vendor/symfony/event-dispatcher/EventDispatcher.php",
            "line": 73,
            "args": [
                [
                    "array",
                    [
                        [
                            "object",
                            "Symfony\\Component\\EventDispatcher\\Debug\\WrappedListener"
                        ],
                        [
                            "object",
                            "Symfony\\Component\\EventDispatcher\\Debug\\WrappedListener"
                        ],
                        [
                            "object",
                            "Symfony\\Component\\EventDispatcher\\Debug\\WrappedListener"
                        ],
                        [
                            "object",
                            "Symfony\\Component\\EventDispatcher\\Debug\\WrappedListener"
                        ]
                    ]
                ],
                [
                    "string",
                    "kernel.view"
                ],
                [
                    "object",
                    "Symfony\\Component\\HttpKernel\\Event\\ViewEvent"
                ]
            ]
        },
        {
            "namespace": "Symfony\\Component\\EventDispatcher",
            "short_class": "EventDispatcher",
            "class": "Symfony\\Component\\EventDispatcher\\EventDispatcher",
            "type": "->",
            "function": "dispatch",
            "file": "/srv/api/vendor/symfony/event-dispatcher/Debug/TraceableEventDispatcher.php",
            "line": 168,
            "args": [
                [
                    "object",
                    "Symfony\\Component\\HttpKernel\\Event\\ViewEvent"
                ],
                [
                    "string",
                    "kernel.view"
                ]
            ]
        },
        {
            "namespace": "Symfony\\Component\\EventDispatcher\\Debug",
            "short_class": "TraceableEventDispatcher",
            "class": "Symfony\\Component\\EventDispatcher\\Debug\\TraceableEventDispatcher",
            "type": "->",
            "function": "dispatch",
            "file": "/srv/api/vendor/symfony/http-kernel/HttpKernel.php",
            "line": 151,
            "args": [
                [
                    "object",
                    "Symfony\\Component\\HttpKernel\\Event\\ViewEvent"
                ],
                [
                    "string",
                    "kernel.view"
                ]
            ]
        },
        {
            "namespace": "Symfony\\Component\\HttpKernel",
            "short_class": "HttpKernel",
            "class": "Symfony\\Component\\HttpKernel\\HttpKernel",
            "type": "->",
            "function": "handleRaw",
            "file": "/srv/api/vendor/symfony/http-kernel/HttpKernel.php",
            "line": 68,
            "args": [
                [
                    "object",
                    "Symfony\\Component\\HttpFoundation\\Request"
                ],
                [
                    "integer",
                    1
                ]
            ]
        },
        {
            "namespace": "Symfony\\Component\\HttpKernel",
            "short_class": "HttpKernel",
            "class": "Symfony\\Component\\HttpKernel\\HttpKernel",
            "type": "->",
            "function": "handle",
            "file": "/srv/api/vendor/symfony/http-kernel/Kernel.php",
            "line": 201,
            "args": [
                [
                    "object",
                    "Symfony\\Component\\HttpFoundation\\Request"
                ],
                [
                    "integer",
                    1
                ],
                [
                    "boolean",
                    true
                ]
            ]
        },
        {
            "namespace": "Symfony\\Component\\HttpKernel",
            "short_class": "Kernel",
            "class": "Symfony\\Component\\HttpKernel\\Kernel",
            "type": "->",
            "function": "handle",
            "file": "/srv/api/public/index.php",
            "line": 25,
            "args": [
                [
                    "object",
                    "Symfony\\Component\\HttpFoundation\\Request"
                ]
            ]
        }
    ]
}
```

</details>

### PR DESCRIPTION
**Pull request by teohhanhui**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tickets       | fixes #1405
| License       | MIT
| Doc PR        | N/A

Cleaner alternative to #1414

### CODE REVIEW

**Comment by teohhanhui** [comment]
WDYT @GregoireHebert?

**Comment by GregoireHebert** [comment]
This is fine to me :)
I think I like your solution better, since it involve less volumes.
I think @dunglas will agree.

File: docker-compose.yml [hunk 1]
```diff
@@ -101,7 +103,7 @@ services:
       - dev-tls
     volumes:
       - ./client:/usr/src/client:rw,cached
-      - dev-certs:/usr/src/client/node_modules/webpack-dev-server/ssl:rw
+      - dev-certs:/usr/src/client/node_modules/webpack-dev-server/ssl:ro,nocopy
```
Comment (teohhanhui) [inline docker-compose.yml]: https://github.com/docker/docker.github.io/issues/2992#issuecomment-383881897

File: api/docker/php/docker-entrypoint.sh [hunk 2]
```diff
@@ -17,6 +17,11 @@ if [ "$1" = 'php-fpm' ] || [ "$1" = 'php' ] || [ "$1" = 'bin/console' ]; then
 	setfacl -R -m u:www-data:rwX -m u:"$(whoami)":rwX var
 	setfacl -dR -m u:www-data:rwX -m u:"$(whoami)":rwX var
 
+	if [ "$APP_ENV" != 'prod' ] && [ -f /certs/localCA.crt ]; then
```
Comment (dunglas) [inline api/docker/php/docker-entrypoint.sh]: Maybe could we add it even in prod as the volume will not be mounted when using K8S and Docker Composer in prod mode.

File: update-deps.sh [hunk 3]
```diff
@@ -20,6 +20,7 @@ sed -i.bak 's;^#TRUSTED_PROXIES=127.0.0.1,127.0.0.2$;TRUSTED_PROXIES=10.0.0.0/8,
 sed -i.bak "s/^#TRUSTED_HOSTS='\^localhost\|example\\\.com\$'$/TRUSTED_HOSTS='^localhost|api$'/" .env
 sed -i.bak 's;^# For a PostgreSQL database, use: "postgresql://db_user:db_password@127.0.0.1:5432/db_name?serverVersion=11"$;# For a MySQL database, use: "mysql://db_user:db_password@127.0.0.1:3306/db_name?serverVersion=5.7";' .env
 sed -i.bak 's;^DATABASE_URL=mysql://db_user:db_password@127.0.0.1:3306/db_name?serverVersion=5.7$;DATABASE_URL=postgres://api-platform:!ChangeMe!@db/api?server_version=12;' .env
+sed -i.bak 's;^MERCURE_PUBLISH_URL=http://mercure/.well-known/mercure$;MERCURE_PUBLISH_URL=https://mercure/.well-known/mercure;' .env
```
Comment (dunglas) [inline update-deps.sh]: We should change the Symfony recipe directly.

Comment (GregoireHebert) [reply update-deps.sh]: you mean in addition to this right?

Comment (teohhanhui) [reply update-deps.sh]: Actually, is `mercure` a good default hostname in the Symfony recipe?

File: api/docker/php/docker-entrypoint.sh (hunk 2 above)
Comment (teohhanhui) [reply api/docker/php/docker-entrypoint.sh]: What purpose does that serve? For local use I'm sure `api/.env` would be preferred over real env vars. Is this relevant for an (automated) e2e testing environment?

File: docker-compose.yml (hunk 1 above)
Comment (GregoireHebert) [reply docker-compose.yml]: I see the point, but @dunglas argued that maybe webpack could not be too happy about this. To be validated.

File: update-deps.sh (hunk 3 above)
Comment (GregoireHebert) [reply update-deps.sh]: I guess this is expected with the current docker configuration.

File: docker-compose.yml (hunk 1 above)
Comment (shaxaaa) [reply docker-compose.yml:106]: This indeed doesn't work well. At some point client and admin can't be started. 
`rm: can't remove '/usr/src/client/node_modules/webpack-dev-server/ssl/server.pem': Read-only file system`

**Pull request merged**